#169445726 replace done by async await in test

ngirimana opened the pull request:
> #### What does this PR do?
* remove done and replace it with async await
> #### Description of Task to be completed
-  remove all debugging statement
-  Replace the done with async-await.
- assert for both data and status codes for every test suit
> #### How should this be manually tested?
* Clone the repo
* run ```cd MyDiary-Adc
* use 'npm install' to install all necessary dependencies
* set the environmental variable in```.env``` file
* Run 'npm test' to run the test
> #### Any background context you want to provide?
* For more details, visit:
   * https://github.com/chaijs/chai-http/issues/156
   * https://jestjs.io/docs/en/asynchronous#async-await
   * https://www.coreycleary.me/know-what-to-test-using-these-recipes-node-service-that-calls-a-database/

> #### What are the relevant pivotal tracker stories?
* [ #169445726](https://www.pivotaltracker.com/story/show/169445726)
> #### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/38111965/67880719-d79ab180-fb47-11e9-8095-f96e02095a43.png)

ngirimana committed:
chore(async await test): replace done by async await

- remove done and replace it with async await

- [Finishes #169445726]

Kasulejoseph reviewed (changes requested):
Hey, well-done on this PR. However, You're not correctly using the `async/await`, you only define the async without the corresponding use of await.. check out this resource https://jestjs.io/docs/en/asynchronous#async-await

Kasulejoseph reviewed (commented):
Follow this thread useful as well https://github.com/chaijs/chai-http/issues/156

ngirimana committed:
chore(await): add await corresponding to async in test

- add await to cope with async

- [Finishes #169445726]

ngirimana committed:
chore(test): remove throw error in test

- remove throw error and test if it will throw the error

- [Finishes #169445726]
